src/modules/mveloKeyServer.js

```diff
@@ -0,0 +1,105 @@
+/**
+ * Copyright (C) 2016-2018 Mailvelope GmbH
+ * Licensed under the GNU Affero General Public License version 3
+ */
+
+/**
+ * @fileOverview A simple HTTP client for Mailvelope Key Server's REST api.
+ */
+
+/** The default URL of the mailvelope authenticating keyserver. */
+const DEFAULT_URL = 'https://keys.mailvelope.com';
+
+let baseUrl = DEFAULT_URL;
+
+/**
+ * Changes the base Key Server URL this module uses.
+ * Default is https://keys.mailvelope.com
+ *
+ * @param {String} value    The server's base url
+ */
+export function setKeyServerURL(value) {
+  baseUrl = value;
+}
+
+/**
+ * Get a verified public key either from the server by either email address,
+ * key id, or fingerprint.
+ * @param {string} options.email         (optional) The user id's email address
+ * @param {string} options.keyId         (optional) The long 16 char key id
+ * @param {string} options.fingerprint   (optional) The 40 char v4 fingerprint
+ * @yield {Object}                       The public key json object
+ */
+export async function lookup(options) {
+  const response = await window.fetch(url(options));
+  if (response.status === 200) {
+    return response.json();
+  }
+}
+
+/**
+ * Upload a public key to the server for verification by the user. Normally
+ * a verification mail is sent out to all of the key's user ids, unless a primary
+ * email attribute is supplied. In which case only one email is sent.
+ * @param {string} options.publicKeyArmored   The ascii armored key block
+ * @yield {undefined}
+ */
+export async function upload({publicKeyArmored}) {
+  const response = await window.fetch(url(), {
+    method: 'POST',
+    headers: new Headers({'Content-Type': 'application/json'}),
+    body: JSON.stringify({publicKeyArmored})
+  });
+  checkStatus(response);
+}
+
+/**
+ * Request deletion of a user's key from the keyserver. Either an email address or
+ * the key id have to be specified. The user will receive a verification email
+ * after the request to confirm deletion.
+ * @param {string} options.email   (optional) The user id's email address
+ * @param {string} options.keyId   (optional) The long 16 char key id
+ * @yield {undefined}
+ */
+export async function remove(options) {
```

Comment:
I didn't find a use of this anywhere. So if it is used I have overlooked it.

Reply:
This is not used at the moment, put a planned feature. So this can stay.